pkg/registry/serving/composer.go

```diff
@@ -0,0 +1,236 @@
+					Name: graphName,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{container},
+					Volumes: []corev1.Volume{
+						{
+							Name: modelSharedMountName,
+							VolumeSource: corev1.VolumeSource{
+								EmptyDir: &corev1.EmptyDirVolumeSource{},
+							},
+						},
+					},
+				},
+			},
+		}
+	}
+
+	composeInitContainer(sdep)
+
+	return nil
+}
+
+// getProbe generate readiness and liveiness
+func getProbe(format, servingName string) *corev1.Probe {
+	path := fmt.Sprintf("/api/status/%v", servingName)
+	if format == string(modeljobsv1alpha1.FormatPMML) {
+		path = fmt.Sprintf("/openscoring/model/%v", servingName)
+	}
+
+	return &corev1.Probe{
+		Handler: corev1.Handler{
+			HTTPGet: &corev1.HTTPGetAction{
+				Path: path,
+				Port: intstr.IntOrString{
+					Type:   intstr.Int,
+					IntVal: 8000,
+				},
+				Scheme: corev1.URISchemeHTTP,
+			},
+		},
+	}
+}
+
+func getModelFormat(pu *seldonv1.PredictiveUnit) string {
+	for _, p := range pu.Parameters {
+		if p.Name == "format" {
+			return p.Value
+		}
+	}
+
+	return ""
+}
+
+func getUserContainerImage(pu *seldonv1.PredictiveUnit) string {
+	format := getModelFormat(pu)
+	if format == string(modeljobsv1alpha1.FormatPMML) {
+		return viper.GetString(envPMMLServingImage)
+	}
+
+	return viper.GetString(envTRTServingImage)
+}
+
+func getRuntimeResource(pu *seldonv1.PredictiveUnit) (*corev1.ResourceRequirements, error) {
+	cpu := ""
+	mem := ""
+	for _, p := range pu.Parameters {
+		if p.Name == "cpu" {
+			cpu = p.Value
+		}
+		if p.Name == "mem" {
+			mem = p.Value
+		}
+	}
+
+	resourcesList := make(corev1.ResourceList)
+	cpuQuantity, err := resource.ParseQuantity(cpu)
+	if err != nil {
+		return nil, err
+	}
+	resourcesList[corev1.ResourceCPU] = cpuQuantity
+
+	memQuantity, err := resource.ParseQuantity(mem)
+	if err != nil {
+		return nil, err
+	}
+	resourcesList[corev1.ResourceMemory] = memQuantity
+	// TODO(@simon-cj): Support GPU resource
+
+	return &corev1.ResourceRequirements{
+		Limits:   resourcesList,
+		Requests: resourcesList,
+	}, nil
+}
+
+func getModelTag(modelUri string) (string, error) {
+	modelURISlice := strings.Split(modelUri, ":")
+	if len(modelURISlice) < 2 {
+		return "", fmt.Errorf("modelUri's format is error")
+	}
+
+	return modelURISlice[len(modelURISlice)-1], nil
+}
+
+func getModelMountPath(servingName string) string {
+	return fmt.Sprintf("/mnt/%v", servingName)
+}
+
+func composeInitContainer(sdep *seldonv1.SeldonDeployment) error {
+	modelMountPath := getModelMountPath(sdep.Name)
+
+	for _, p := range sdep.Spec.Predictors {
+		// simple model serving, the number of ComponentSpecs is 1
+		if len(p.ComponentSpecs) != 1 {
+			return fmt.Errorf("too many or too less componentspecs")
+		}
+
+		container := corev1.Container{
+			Name:  "model-initializer",
+			Image: viper.GetString(envModelInitializerImage),
+			Args:  []string{p.Graph.ModelURI, modelMountPath},
+			Env: []corev1.EnvVar{
+				{
+					Name:  "AWS_ACCESS_KEY_ID",
```

Comment:
Please add some comments about the KEY to let readers know why we use AWS_ACCESS_KEY_ID instead of ORMB_XXX.